tiledb/sm/query_plan/test/unit_query_plan.cc

```diff
@@ -0,0 +1,350 @@
+/**
+ * @file tiledb/sm/query_plan/test/unit_query_plan.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines a test `main()`
+ */
+
+#include <test/support/tdb_catch.h>
+
+#include "tiledb/api/c_api/query_plan/query_plan_api_external_experimental.h"
+
+#include <tiledb/sm/c_api/tiledb_struct_def.h>
+#include <string>
+#include "test/support/src/helpers.h"
+#include "test/support/src/vfs_helpers.h"
+#ifdef _WIN32
+#include "tiledb/sm/filesystem/win.h"
+#else
+#include "tiledb/sm/filesystem/posix.h"
+#endif
+
+#include "tiledb/sm/filesystem/uri.h"
+
+#include "external/include/nlohmann/json.hpp"
+
+using namespace tiledb::test;
+
+struct QueryPlanFx {
+  // TileDB context and vfs
+  tiledb_ctx_t* ctx_;
+  tiledb_vfs_t* vfs_;
+
+  // Vector of supported filsystems
+  const std::vector<tdb_unique_ptr<SupportedFs>> fs_vec_;
+
+  // Functions
+  QueryPlanFx();
+  ~QueryPlanFx();
+  void remove_temp_dir(const std::string& path);
+  void create_temp_dir(const std::string& path);
+  void create_dense_array(const std::string& path);
+  void create_sparse_array(const std::string& path);
+};
+
+QueryPlanFx::QueryPlanFx()
+    : fs_vec_(tdb_vfs_test_get_fs_vec()) {
+  // Initialize vfs test
+  REQUIRE(tdb_vfs_test_init(fs_vec_, &ctx_, &vfs_).ok());
+}
+
+QueryPlanFx::~QueryPlanFx() {
+  // Close vfs test
+  REQUIRE(tdb_vfs_test_close(fs_vec_, ctx_, vfs_).ok());
+  tiledb_vfs_free(&vfs_);
+  tiledb_ctx_free(&ctx_);
+}
+
+void QueryPlanFx::create_temp_dir(const std::string& path) {
```

Comment:
Use `TemporaryDirectoryFixture` here.

Reply:
Nice, we should document the existence of this, it saves a lot of time knowing this exists!